### PR TITLE
Resolved full name of release author in release blog

### PR DESCRIPTION
**Pull request by phillipj**
In our release blog posts, we want to display the full name of the dev who cut the release, instead of the github login such as `rvagg` or `Fishrock123`. This was proposed by @rvagg in [#186 comment](https://github.com/nodejs/new.nodejs.org/pull/186#issuecomment-142472144).

These changes fixes that by resolving the full name via github's API, based on their github login specified in the changelog.

Also did some refactoring of some hairy changelog regex matching, to something a bit less complex and readable.

Refs #156.

/cc @nodejs/release

### CODE REVIEW

**Comment by mikeal** [comment]
Looks like there's a failure in the slug test.

**Comment by phillipj** [comment]
> Looks like there's a failure in the slug test.

Great! Didnt touch that code at all, and it passed after a couple of Travis restarts. Dont see any apparent reason for it to be unstable, we'll have keep an eye on that slug test.

**Comment by mikeal** [comment]
weird, should probably log another issue for that test being flaky.

File: scripts/release-post.js [hunk 1]
```diff
@@ -71,41 +72,56 @@ function findLatestVersion (cb) {
...
+    .then((author) => download(`https://api.github.com/users/${author}`))
+    .then(JSON.parse)
+    .then((githubRes) => githubRes.name)
+}
+
 function fetchChangelog (version) {
+  // matches a complete release section,
+  // support release sections with headers like:
+  // ## 2015-09-22, Version 4.1.1 (Stable), @rvagg
+  const rxSection = new RegExp(`## \\d{4}-\\d{2}-\\d{2}, Version ${version} \\(Stable\\)[\\s\\S]*?(?=## \\d{4})`)
```
Comment (rvagg) [inline scripts/release-post.js]: We're going to need to extend this, it'll include "LTS" and "Maintenance" too. Perhaps just `\\([^\\)]+\\)` would do.

Comment (phillipj) [inline scripts/release-post.js]: Sure thing! Just pushed an update.

**Comment by phillipj** [comment]
@nodejs/release I just pushed an update to this PR, supporting version policy and codename aswell, as a response to #239.

Also removed version policy from the blog post slug, as I dont see the big value of it, other adding more complexity into this script. Agreed @nodejs/website?

**Comment by phillipj** [comment]
Merging this now so author will be part of the next release post.

**Pull request merged**